scripts/validate_metadata.py

```diff
@@ -0,0 +1,238 @@
+            kfp_present = any(d.get('name') == 'Pipelines' for d in kf_dependencies)
+            if not kfp_present:
+                raise ValueError("metadata.yaml for component '{}' is missing Kubeflow Pipelines dependency.".format(component_name))
+
+            # Dependency versions must be correctly formatted by semantic versioning.
+            invalid_dependencies = get_invalid_versions(kf_dependencies) + get_invalid_versions(ext_dependencies)
+            if len(invalid_dependencies) > 0:
+                raise ValueError("metadata.yaml for component '{}' contains one or more dependencies with invalid semantic versioning: {}.".format(component_name, invalid_dependencies))
+
+        elif field == "tags":
+            tags_val = metadata.get("tags")
+            if not (isinstance(tags_val, list)):
+                raise ValueError("{} value identified in field 'tags' in metadata.yaml for component '{}'. Value must be string array.".format(value_type, component_name))
+            if not all(isinstance(item, str) for item in tags_val):
+                raise ValueError("The following tags in metadata.yaml for component '{}': {}. Expected an array of scalar strings.".format(component_name, tags_val))
+        elif field == "ci":
+            ci_val = metadata.get("ci")
+            if not isinstance(ci_val, dict):
+                raise ValueError("{} value identified for field 'ci' in metadata.yaml for component '{}'. Value must be dictionary.".format(value_type, component_name))
+            keys = set(ci_val.keys())
+            if not (keys == {"skip_dependency_probe"}):
+                raise ValueError("The following field(s) were found in field 'ci' in metadata.yaml for component '{}': {}. Only 'skip_dependency_probe' is valid.".format(component_name, list(ci_val.keys())))
+            probe = ci_val.get("skip_dependency_probe")
+            if probe is not None and not isinstance(probe, bool):
+                raise ValueError("metadata.yaml expects a boolean value for skip_dependency_probe but {} value provided: '{}'.".format(type(probe).__name__, probe))
+
+        elif field == "links":
+            links_value = metadata.get("links")
+            if not isinstance(links_value, dict):
+                raise ValueError("{} value identified in field 'links' in metadata.yaml for component '{}'. Value must be dictionary.".format(value_type, component_name))
+    return True
+
+"""
+Validate that OWNERS file contains at least one approver."""
+def validate_owners_file(filepath: Path) -> bool:
+    with open(filepath) as f:
+        for line, next_line in pairwise(f):
+            if line.startswith("approvers:") and next_line.startswith("-") and len(next_line) > 2:
+                return True
+        return False
+
+"""
+Validate that both OWNERS and metadata.yaml files are present in the directory for the input component."""
+def validate_metadata_files_in_dir(component_directory: str) -> bool:
```

Comment:
nit: returning a bool doesn't seem like it's doing much since you never actually return false (same with `validate_metadata_yaml`). Since basically these functions only ever do "raise exception or return True" you can just assume that if the function succeeded, validation succeeded and not have to return a bool 

or you could change your approach  and log validation failures and return "false" upon fail rather than raise exceptions, either approach is fine to me